core/uploader.go

```diff
@@ -182,12 +223,28 @@ func extractThumb(session drivers.OSSession, filename string, segment []byte) er
 		return fmt.Errorf("opening file failed: %w", err)
 	}
 	defer f.Close()
-	_, err = session.SaveData(context.Background(), "../latest.jpg", f, &drivers.FileProperties{
-		CacheControl: "max-age=5",
-		Metadata:     expiryField,
-	}, 10*time.Second)
+	thumbData, err := io.ReadAll(f)
+	if err != nil {
+		return fmt.Errorf("failed to read file: %w", err)
+	}
+
+	thumbURL := outputURI.JoinPath("../latest.jpg")
+	fields := &drivers.FileProperties{CacheControl: "max-age=5"}
+	_, _, err = uploadFileWithBackup(thumbURL, thumbData, fields, 10*time.Second, true, storageBackupURLs)
 	if err != nil {
 		return fmt.Errorf("saving thumbnail failed: %w", err)
 	}
 	return nil
 }
+
+func newFileProperties(base *drivers.FileProperties) *drivers.FileProperties {
```

Comment:
Do you mean we should just include that `expiryField` on every storage? I did this cause I didn't know the implications of that metadata in other storages so preferred to stay on the safe side – with the same logic as before – of only including this metadata on the Storj files. 

The helper itself only showed up because I wanted to avoid this code snippet which is very verbose in Go for very few real logic:
```go
var newFields drivers.FileProperties{}
if fields != nil {
	newFields := *fields
}
metadata := map[string]string{}
for k, v := range fields.Metadata {
	metadata[k] = v
}
for k, v := range expiryField {
	metadata[k] = v
}
newFields.Metadata = metadata
fields = &newFields
```

An intermediate simplification would be assuming that we never use `Metadata` for anything else, so we only need to copy the struct and then set the field. We still need to copy the struct tho not to modify the input and have the different uploads affecting each other. If we do that, we can drop the helper and have:

```go
var newFields drivers.FileProperties{}
if fields != nil {
	newFields := *fields
}
newFields.Metadata = expiryFields
fields = &newFields
```

I'll follow that approach, let me know if you prefer some other way.